[BUG]: Invalid interface conversion on github_team.parent_team_id

### Describe the need

There appears to be an invalid interface conversion on version `v5.24.0` of github_team.parent_team_id which results in a crash when applying a resource update.

It appears to be because of this :
https://github.com/integrations/terraform-provider-github/blob/872e253865a3e7eb2656d96b144d969062329a90/github/resource_github_team.go#L226

The bug seems to have been introduced by https://github.com/integrations/terraform-provider-github/pull/1664 , they overlooked a type assertion

### SDK Version

_No response_

### API Version

_No response_

### Relevant log output

```shell
Stack trace from the terraform-provider-github_v5.24.0 plugin:

panic: interface conversion: interface {} is string, not int

goroutine 38 [running]:
github.com/integrations/terraform-provider-github/v5/github.resourceGithubTeamUpdate(0xc000e32d20, {0x1c47d40?, 0xc000618280})
        github.com/integrations/terraform-provider-github/v5/github/resource_github_team.go:226 +0x868
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0004db2c0, 0xc00094a4b0, 0xc0005b1060, {0x1c47d40, 0xc000618280})
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:326 +0x3e3
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000142d00, 0xc000b2f8c8, 0x1e997d2?, 0xf?)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/provider.go:294 +0x70
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00027c680, {0xc000e32070?, 0x10b3d06?}, 0xc000e32070)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:895 +0x7c5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1e428c0?, 0xc00027c680}, {0x21e73c0, 0xc0003b2b10}, 0xc000e32000, 0x0)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3305 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00022e000, {0x21eb308, 0xc000783040}, 0xc0006ee000, 0xc0003b32c0, 0x2a0a7a0, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc00022e000, {0x21eb308, 0xc000783040}, 0xc0006ee000, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x28a
panic: interface conversion: interface {} is string, not int

goroutine 39 [running]:
github.com/integrations/terraform-provider-github/v5/github.resourceGithubTeamUpdate(0xc0005b98f0, {0x1c47d40?, 0xc000618280})
        github.com/integrations/terraform-provider-github/v5/github/resource_github_team.go:226 +0x868
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0004db2c0, 0xc0008001e0, 0xc000caac00, {0x1c47d40, 0xc000618280})
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:326 +0x3e3
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000142d00, 0xc0002d38c8, 0x1e997d2?, 0xf?)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/provider.go:294 +0x70
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00027c680, {0xc0005b8e00?, 0x10b3d06?}, 0xc0005b8e00)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:895 +0x7c5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1e428c0?, 0xc00027c680}, {0x21e73c0, 0xc0007a4030}, 0xc0005b8d90, 0x0)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3305 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00022e000, {0x21eb308, 0xc000783040}, 0xc0006ee240, 0xc0003b32c0, 0x2a0a7a0, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc00022e000, {0x21eb308, 0xc000783040}, 0xc0006ee240, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

Error: The terraform-provider-github_v5.24.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/github/resource_github_team.go
+++ b/github/resource_github_team.go
@@ -223,8 +223,11 @@ func resourceGithubTeamUpdate(d *schema.ResourceData, meta interface{}) error {
 		Privacy:     github.String(d.Get("privacy").(string)),
 	}
 	if parentTeamID, ok := d.GetOk("parent_team_id"); ok {
-		id := int64(parentTeamID.(int))
-		editedTeam.ParentTeamID = &id
+		teamId, err := getTeamID(parentTeamID.(string), meta)
+		if err != nil {
+			return err
+		}
+		editedTeam.ParentTeamID = &teamId
 	}
 
 	teamId, err := strconv.ParseInt(d.Id(), 10, 64)

